plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/TestIcebergMetadataFileOperations.java

```diff
@@ -75,90 +83,220 @@ protected DistributedQueryRunner createQueryRunner()
+
+        public FileOperation(FileType fileType, OperationType operationType)
         {
-            this.allCounts = requireNonNull(allCounts, "allCounts is null");
+            this.fileType = requireNonNull(fileType, "fileType is null");
+            this.operationType = requireNonNull(operationType, "operationType is null");
         }
 
-        public OperationCounts forQueryId(String queryId)
+        public FileOperation(OperationContext operationContext)
         {
-            this.queryId = Optional.of(queryId);
-            return this;
+            this(fromFilePath(operationContext.getFilePath()), operationContext.getOperationType());
         }
 
-        public OperationCounts forPathContaining(String value)
+        @Override
+        public boolean equals(Object o)
         {
-            this.pathPredicate = path -> path.contains(value);
-            return this;
+            if (this == o) {
+                return true;
+            }
+            if (o == null || getClass() != o.getClass()) {
+                return false;
+            }
+            FileOperation that = (FileOperation) o;
+            return fileType == that.fileType && operationType == that.operationType;
         }
 
-        public OperationCounts forOperation(TrackingFileIoProvider.OperationType type)
+        @Override
+        public int hashCode()
         {
-            this.type = Optional.of(type);
-            return this;
+            return Objects.hash(fileType, operationType);
         }
 
-        public Map<OperationContext, Integer> get()
+        @Override
+        public String toString()
         {
-            return allCounts.entrySet().stream()
-                    .filter(entry -> queryId.map(id -> entry.getKey().getQueryId().equals(id)).orElse(true))
-                    .filter(entry -> pathPredicate.test(entry.getKey().getFilePath()))
-                    .filter(entry -> type.map(value -> value == entry.getKey().getOperationType()).orElse(true))
-                    .collect(toImmutableMap(Map.Entry::getKey, Map.Entry::getValue));
+            final StringBuilder sb = new StringBuilder("FileOperation{");
+            sb.append("fileType=").append(fileType);
+            sb.append(", operationType=").append(operationType);
+            sb.append('}');
+            return sb.toString();
         }
+    }
+
+    enum FileType
+    {
+        METADATA,
+        SNAPSHOT,
+        MANIFEST;
 
-        public int sum()
+        public static FileType fromFilePath(String path)
         {
-            return get().values().stream().mapToInt(Integer::intValue).sum();
+            if (path.contains("metadata.json")) {
+                return METADATA;
+            }
+            if (path.contains("/snap-")) {
+                return SNAPSHOT;
+            }
+            if (path.contains("-m0.avro")) {
+                return MANIFEST;
+            }
+            throw new IllegalArgumentException("File not recognized");
```

Comment:
report path, so that whenever this fails, it's easier to change the code to accomodate

```suggestion
            throw new IllegalArgumentException("File not recognized: " + path);
```